config/core/resources/subscription.yaml

```diff
@@ -127,6 +127,7 @@ spec:
                       namespace:
                         description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/ This is optional field, it gets defaulted to the object holding it if left out.'
                         type: string
+                    x-kubernetes-preserve-unknown-fields: true # This is necessary to enable the experimental feature
```

Comment:
I'm usually not a big fan on this. The reason is what happen if, let say, I type `subcriber` instead of `subscriber`? Without `x-kubernetes-preserve-unknown-fields: true`, I would get an error which is the expected behavior. 

Maybe the validation webhook checks for unknown fields, but last time *I* checked it didn't.

If that the case, then I would rather explicit add the new field and indicate in the description it is experimental.

Reply:
> Maybe the validation webhook checks for unknown fields, but last time I checked it didn't.

It should fail because the mapping to the object fails, right?

> The reason is what happen if, let say, I type subcriber instead of subscriber? 

It will fail because I've added the preserve unknown specifically to the `ref` sub-object.

> If that the case, then I would rather explicit add the new field and indicate in the description it is experimental.

I think this was discussed before in the experimental features proposal and during an eventing wg meeting too :smile:

* Either we add the schema field, risking to make that too explicit and creating problems, like breaking clusters in future
* Or we add the preserve unknown, having more liberty to change things without breaking the cluster 

Reference: https://docs.google.com/document/d/1AoH0FyLeuHIg5snrlCKzELQv-NEA6bbjTU2Qv3zlW5k/edit?disco=AAAAIPGZ3P8

Reply:
> It will fail because I've added the preserve unknown specifically to the ref sub-object.

right, wrong example. `namepace` vs `namespace` then.

> I think this was discussed before in the experimental features proposal and during an eventing wg meeting too

That what I thought but wanted to give another try here. 

Anyway, I least I expressed my opinion...

Reply:
@lionelvillard just tested to apply this one:

```
apiVersion: messaging.knative.dev/v1
kind: Subscription
metadata:
  name: example-subscription
spec:
  subscriber:
    ref:
      apiVersion: serving.knative.dev/v1
      kind: Service
      name: example-sink
      someotherstuff: blablabla
```

And it failed with this:

```
Error from server (BadRequest): error when creating "sample.yaml": admission webhook "webhook.eventing.knative.dev" denied the request: mutation failed: cannot decode incoming new object: json: unknown field "someotherstuff"
```

So I guess the typo thing shouldn't be problematic

Reply:
actually the error is first raised by the defaulting webhook, but same. Found where: https://github.com/knative/pkg/blob/main/webhook/resourcesemantics/defaulting/defaulting.go#L246

All good! 

Thanks!